readalongs/web_api.py

```diff
@@ -203,5 +204,174 @@ def create_grammar(xml):
     return dict_data, fsg_data, text_data
 
 
+class ConvertRequest(BaseModel):
+    """Convert Request contains the RAS-processed XML and SMIL alignments"""
+
+    audio_duration: float = Field(
+        example=2.01,
+        gt=0.0,
+        title="The duration of the audio used to create the alignment, in seconds.",
+    )
+
+    output_format: str = Field(
+        example="TextGrid",
+        regex="^(?i)(eaf|TextGrid|srt|vtt)$",
+        title="Format to convert to, one of TextGrid (Praat), eaf (ELAN), srt (SRT subtitles), or vtt (VTT subtitles).",
+    )
+
+    xml: str = Field(
+        title="The processed_xml returned by /assemble.",
+        example=dedent(
+            """\
+            <?xml version='1.0' encoding='utf-8'?>
+            <TEI>
+                <text xml:lang="dan" fallback-langs="und" id="t0">
+                    <body id="t0b0">
+                        <div type="page" id="t0b0d0">
+                            <p id="t0b0d0p0">
+                                <s id="t0b0d0p0s0"><w id="t0b0d0p0s0w0" ARPABET="HH EH Y">hej</w> <w id="t0b0d0p0s0w1" ARPABET="V Y D EH N">verden</w></s>
+                            </p>
+                        </div>
+                    </body>
+                </text>
+            </TEI>"""
+        ),
+    )
+
+    smil: str = Field(
+        title="The result of aligning xml to the audio with SoundSwallower(.js)",
+        example=dedent(
+            """\
+            <smil xmlns="http://www.w3.org/ns/SMIL" version="3.0">
+                <body>
+                    <par id="par-t0b0d0p0s0w0">
+                        <text src="hej-verden.xml#t0b0d0p0s0w0"/>
+                        <audio src="hej-verden.mp3" clipBegin="0.14" clipEnd="0.78"/>
+                    </par>
+                    <par id="par-t0b0d0p0s0w1">
+                        <text src="hej-verden.xml#t0b0d0p0s0w1"/>
+                        <audio src="hej-verden.mp3" clipBegin="0.78" clipEnd="1.89"/>
+                    </par>
+                </body>
+            </smil>"""
+        ),
+    )
+
+
+class ConvertResponse(BaseModel):
+    """Convert response has the requesed converted file's contents"""
+
+    file_name: str = Field(
+        title="A suggested name for this output file: aligned + the standard extension"
+    )
+
+    file_contents: str = Field(
+        title="Full contents of the converted file in the format requested."
+    )
+
+    other_file_name: Union[None, str] = Field(
+        title="Suggested name for the second file, if the conversion generates two files"
+    )
+
+    other_file_contents: Union[None, str] = Field(
+        title="Full contents of the second file, if any"
+    )
+
+
+def slurp_file(filename):
+    """Slurp a file into one string"""
+    with open(filename, "r", encoding="utf-8") as f:
+        return f.read()
+
+
+@v1.post("/convert_alignment", response_model=ConvertResponse)
```

Comment:
```suggestion
@v1.post("/convert_alignment/{output_format}", response_model=ConvertResponse)
async def convert_alignment(input: ConvertRequest, output_format) -> ConvertResponse:
```

This is how you do what you were suggesting. I would also kind of prefer this I think instead of bundling it with the body of the request? I mean it's certainly what you would do in a RESTful API, but this is more RPC-ish. 

See "path parameters" https://fastapi.tiangolo.com/tutorial/path-params/
Another option (and what might make sense for specifying sentences vs words on vtt would be something like:

`/convert_alignment/vtt?tier=sentences` and in that case the `tier=` would be a "query parameter" (https://fastapi.tiangolo.com/tutorial/query-params/). Then we could create an Enumaration class for the output formats that we accept and let FastAPI handle the case where somebody requests something else. See https://fastapi.tiangolo.com/tutorial/path-params/?h=enum#working-with-python-enumerations

Reply:
Hum, I like the path parameter, but I can't figure out how to keep my documentation I had for the enum values.
With `output_name` as a member of `ConvertRequest`, as an enum or a str, I can do:
```
output_format: FormatName = Field(
    example=FormatName.textgrid,
    title="Format to convert to, one of textgrid (Praat TextGrid), eaf (ELAN EAF), srt (SRT subtitles), or vtt (VTT subtitles).",
)
```

Is there a corresponding place I can put that `title` info when I do a path parameter? I tried giving the path param a default value as a Field with such a title, but it didn't put that title in the documentation. 

I guess there's always the docstring for the endpoint itself, that would do too.

Most satisfying, though, would be if my `class FormatName(Enum)` could itself have its own `title` or similar documentation...